### PR TITLE
Standardize code documentation in logic/* - Closes #1518 

### LINKED ISSUE #1518: Apply code documentation standards in logic/*
Parent: #1333

### PR DESCRIPTION
**Pull request by Tschakki**
### What was the problem?
Code documentation standards are not applied consistently in the project.

### How did I fix it?
Apply the same JSDoc standards in every file.

### How to test it?
`npm run docs:build`
`npm run docs:serve`

### Review checklist

* The PR solves #1518 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: logic/peers.js [hunk 1]
```diff
@@ -210,9 +227,10 @@ Peers.prototype.remove = function(peer) {
 };
 
 /**
- * Returns private list of peers
+ * Returns private list of peers.
+ *
  * @param {boolean} [normalize] - If true transform list to object
```
Comment (willclarktech) [inline logic/peers.js]: As above.

File: logic/round.js [hunk 2]
```diff
@@ -107,7 +110,9 @@ Round.prototype.mergeBlockGenerator = function() {
 
 /**
  * If outsiders content, calls sql updateMissedBlocks.
- * @return {}
+ *
+ * @returns {}
```
Comment (willclarktech) [inline logic/round.js]: What does this mean? Elsewhere we either put something inside the braces or leave out the tag entirely (maybe with a todo to add it later), don't we?

File: logic/round.js [hunk 3]
```diff
@@ -122,18 +127,19 @@ Round.prototype.updateMissedBlocks = function() {
...
 	return (this.t || this.scope.library.db).rounds.getVotes(this.scope.round);
 };
 
 /**
  * Calls getVotes with round.
- * @implements {getVotes}
- * @implements {modules.accounts.generateAddressByPublicKey}
- * @return {function} Promise
+ *
+ * @returns {function} Promise
```
Comment (willclarktech) [inline logic/round.js]: This is confusing.

File: logic/transaction.js [hunk 4]
```diff
@@ -256,11 +268,13 @@ Transaction.prototype.getBytes = function(
...
- * Calls `ready` based on transaction type (see privateTypes)
+ * Calls `ready` based on transaction type (see privateTypes).
+ *
  * @see privateTypes
- * @param {transaction} transaction
- * @param {account} sender
- * @return {function|boolean} calls `ready` | false
+ * @param {transaction} transaction - Description of the param
+ * @param {account} sender - Description of the param
+ * @returns {function|boolean} calls `ready` | false
```
Comment (willclarktech) [inline logic/transaction.js]: This is confusing: it returns the result of calling the function, not the function itself. Unless it's a HOF?

File: logic/transaction.js [hunk 5]
```diff
@@ -422,13 +442,14 @@ Transaction.prototype.process = function(
 /**
  * Validates parameters.
  * Calls `process` based on transaction type (see privateTypes)
```
Comment (willclarktech) [inline logic/transaction.js]: Full stop.

File: logic/transaction.js [hunk 6]
```diff
@@ -964,15 +983,14 @@ Transaction.prototype.undo = function(transaction, block, sender, cb) {
  * unconfirmed balance negative amount.
  * Calls `applyUnconfirmed` based on transaction type (privateTypes). If error merge
  * account with amount.
+ *
  * @see privateTypes
```
Comment (willclarktech) [inline logic/transaction.js]: Is there a link we can provide for this?

File: logic/transaction_pool.js [hunk 7]
```diff
@@ -136,8 +142,9 @@ TransactionPool.prototype.transactionInPool = function(id) {
 
 /**
  * Gets an unconfirmed transaction based on transaction id.
- * @param {string} id - Transaction id.
- * @return {(Object|undefined)} - Transaction or undefined.
+ *
+ * @param {string} id - Transaction id
+ * @returns {(Object|undefined)} - Transaction or undefined
```
Comment (willclarktech) [inline logic/transaction_pool.js]: What do the round brackets signify?

File: logic/transaction_pool.js [hunk 8]
```diff
@@ -107,9 +111,10 @@ function TransactionPool(
 // Public methods
 /**
  * Bounds input parameters to private variable modules.
- * @param {Accounts} accounts - Accounts module instance.
- * @param {Transactions} transactions - Transactions module instance.
- * @param {Loader} loader - Loader module instance.
+ *
+ * @param {Accounts} accounts - Accounts module instance
```
Comment (willclarktech) [inline logic/transaction_pool.js]: Are we consistent in our use of lowercase and uppercase for `accounts`, `transactions` etc?

File: logic/transaction_pool.js [hunk 9]
```diff
@@ -464,13 +483,9 @@ TransactionPool.prototype.reindexQueues = function() {
...
- * @implements {getBundledTransactionList}
- * @implements {removeBundledTransaction}
- * @implements {processVerifyTransaction}
- * @implements {removeUnconfirmedTransaction}
- * @implements {queueTransaction}
- * @param {function} cb - Callback function.
- * @return {setImmediateCallback} err | cb
+ *
+ * @param {function} cb - Callback function
+ * @returns {setImmediateCallback} err | cb
```
Comment (willclarktech) [inline logic/transaction_pool.js]: This is different to `err, transactions` above. Which do we want? Does it even make sense?

File: logic/transfer.js [hunk 10]
```diff
@@ -247,8 +271,10 @@ Transfer.prototype.objectNormalize = function(transaction) {
 
 /**
  * Checks if asset exists, if so, returns value, otherwise returns null.
- * @param {Object} raw
- * @return {transferAsset|null}
+ *
+ * @param {Object} raw - Description of the param
+ * @returns {transferAsset|null}
```
Comment (willclarktech) [inline logic/transfer.js]: If we're using `!` and `?` we could use `?` here.

File: modules/blocks/utils.js [hunk 11]
```diff
@@ -66,8 +66,8 @@ function Utils(
  * Normalize blocks and their transactions
  *
  * @method readDbRows
- * @param  {[Object]} rows Data from full_blocks_list view
- * @return {[Object]} blocks Normalized list of blocks with transactions
+ * @param  {Array} rows Data from full_blocks_list view
+ * @return {Array} blocks Normalized list of blocks with transactions
```
Comment (willclarktech) [inline modules/blocks/utils.js]: Is this related to this PR?

File: logic/account.js [hunk 12]
```diff
@@ -52,7 +61,7 @@ function Account(db, schema, logger, cb) {
 
 	this.table = 'mem_accounts';
 	/**
-	 * @typedef {Object} account
+	 * @typedef {Object} account_model
```
[the remark this answers is not in the document]
Comment (Tschakki) [reply logic/account.js]: The typedef makes it possible to add a property to the class that has this particular typedef as type, like this: 
`@property {account_model} model`
`@property {account_schema} schema`
Do you know a better way of describing objects and including them into the JSDoc html presentation?

[the remark this answers is not in the document]
Comment (willclarktech) [reply logic/account.js]: Maybe it would make more sense to locate it in a typedefs file? Or at the top of the file?

[the remark this answers is not in the document]
Comment (Tschakki) [reply logic/account.js]: Why not right above the object that it is describing?

File: modules/blocks/utils.js (hunk 11 above)
Comment (Tschakki) [reply modules/blocks/utils.js]: oops, included it by accident. 

It fixes a bug, because it resulted in an error when trying to build the JSDoc HTML files.
Can I leave it here or should it be part of the PR for the modules-folder (coming soon)?

File: logic/multisignature.js [hunk 13]
```diff
@@ -274,21 +287,24 @@ Multisignature.prototype.verify = function(transaction, sender, cb) {
...
 };
 
 /**
  * Returns a buffer with bytes from transaction asset information.
+ *
  * @requires bytebuffer
  * @see {@link https://github.com/dcodeIO/bytebuffer.js/wiki/API}
- * @param {transaction} transaction - Uses multisignature from asset.
+ * @param {transaction} transaction - Uses multisignature from asset
  * @returns {!Array} Contents as an ArrayBuffer.
```
[the remark this answers is not in the document]
Comment (Tschakki) [reply logic/multisignature.js]: From JSDoc documentation: `Indicates that the value is of the specified type, but cannot be null.`

File: logic/broadcaster.js [hunk 14]
```diff
@@ -138,10 +148,12 @@ Broadcaster.prototype.getPeers = function(params, cb) {
...
 /**
- * Adds new object {params, options} to queue array .
- * @param {Object} params
- * @param {Object} options
- * @return {Object[]} queue private variable with new data
+ * Adds new object {params, options} to queue array.
+ *
+ * @param {Object} params - Description of the param
+ * @param {Object} options - Description of the param
+ * @returns {Object[]} queue private variable with new data
```
[the remark this answers is not in the document]
Comment (Tschakki) [reply logic/broadcaster.js]: Good question, both syntax are valid, see: http://usejsdoc.org/tags-type.html
Do you have a preference? 
If `Object[]` is only used here, I would change it to `Array.<Object>`, but I don't have a strong opinion on it.

File: logic/in_transfer.js [hunk 15]
```diff
@@ -117,9 +130,11 @@ InTransfer.prototype.process = function(transaction, sender, cb) {
 /**
  * Creates buffer with inTransfer content:
  * - dappId
- * @param {transaction} transaction
- * @return {Array} Buffer
+ *
+ * @param {transaction} transaction - Description of the param
+ * @returns {Array} Buffer
  * @throws {e} Error
```
[the remark this answers is not in the document]
Comment (Tschakki) [reply logic/in_transfer.js]: What do you mean?

File: logic/multisignature.js [hunk 16]
```diff
@@ -274,21 +287,24 @@ Multisignature.prototype.verify = function(transaction, sender, cb) {
...
 /**
  * Returns transaction with setImmediate.
- * @param {transaction} transaction
- * @param {account} sender
+ *
+ * @param {transaction} transaction - Description of the param
+ * @param {account} sender - Description of the param
  * @param {function} cb - Callback function.
  * @returns {setImmediateCallback} Null error
  * @todo check extra parameter sender.
```
[the remark this answers is not in the document]
Comment (Tschakki) [reply logic/multisignature.js]: What do you mean?

File: logic/peer.js [hunk 17]
```diff
@@ -130,9 +138,11 @@ Peer.prototype.normalize = function(peer) {
 
 /**
  * Checks number or assigns default value from parameter.
- * @param {number} integer
- * @param {number} [fallback]
- * @return {number} if not integer returns fallback
+ *
+ * @param {number} integer - Description of the param
+ * @param {number} [fallback] - Description of the param
```
[the remark this answers is not in the document]
Comment (Tschakki) [reply logic/peer.js]: It indicates that the parameter is optional

File: logic/peers.js (hunk 1 above)
Comment (Tschakki) [reply logic/peers.js]: It indicates that the parameter is optional

File: logic/round.js (hunk 2 above)
Comment (Tschakki) [reply logic/round.js]: I don't know, it was already documented like this. I added a `@todo` that it should be described better. But maybe it is better to remove it completely and to add a new todo, I will do this with the next commit :)

File: logic/round.js (hunk 3 above)
Comment (Tschakki) [reply logic/round.js]: Should I add a todo here?

File: logic/transaction.js (hunk 4 above)
Comment (Tschakki) [reply logic/transaction.js]: Maybe it makes sense to review existing contents of the docs again, this PR focuses more on introducing consistent code styles. I'm not really reviewing types of values, as long as there is a type defined, I assume it is the right type.

File: logic/transaction.js (hunk 6 above)
Comment (Tschakki) [reply logic/transaction.js]: Yes, will be included in the next commits :)

File: logic/transaction_pool.js (hunk 8 above)
Comment (Tschakki) [reply logic/transaction_pool.js]: No, probably not, but there are also no corresponding typedefs, yet.

File: logic/transaction_pool.js (hunk 7 above)
Comment (Tschakki) [reply logic/transaction_pool.js]: I don't think they are needed. 
Will be removed with next commits.

File: logic/transaction_pool.js (hunk 9 above)
Comment (Tschakki) [reply logic/transaction_pool.js]: As mentioned above, I think after standardization it makes sense to revisit the docs and complete/ correct them.
I'm not sure this is part of this PR, but we could leave `@todo`s for everything that should be looked over at again.

File: logic/in_transfer.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (willclarktech) [reply logic/in_transfer.js]: The type is `Error`, no?

File: logic/multisignature.js (hunk 16 above)
[the remark this answers is not in the document]
Comment (willclarktech) [reply logic/multisignature.js]: Is this still something that needs to be done or can we remove the line?

File: logic/multisignature.js (hunk 13 above)
[the remark this answers is not in the document]
Comment (willclarktech) [reply logic/multisignature.js]: Right, but that's presumably the case for a lot of our types in other cases. So far we haven't established a consistent pattern for nullable/optional/non-nullable types.

File: logic/peer.js (hunk 17 above)
[the remark this answers is not in the document]
Comment (willclarktech) [reply logic/peer.js]: This feeds into the comment above.

File: logic/round.js (hunk 3 above)
Comment (willclarktech) [reply logic/round.js]: Sounds like a good idea.

File: logic/account.js (hunk 12 above)
[the remark this answers is not in the document]
Comment (willclarktech) [reply logic/account.js]: This goes back to my original confusion. `this.model` doesn't match that type, it's an array of objects. What is this type for?

File: logic/broadcaster.js (hunk 14 above)
[the remark this answers is not in the document]
Comment (willclarktech) [reply logic/broadcaster.js]: I'd be tempted to get rid of nesting altogether and just use `Array` (and revisit later). But if this is the only place we use `[]` it would be relatively easy to just switch to `<>`.

File: logic/account.js (hunk 12 above)
[the remark this answers is not in the document]
Comment (Tschakki) [reply logic/account.js]: I get your point now. I think we don't need the typedef here, will be removed with next commit.

File: logic/broadcaster.js (hunk 14 above)
[the remark this answers is not in the document]
Comment (Tschakki) [reply logic/broadcaster.js]: As discussed, we will open a new issue for standardizing typedefs in all files

File: logic/dapp.js [hunk 18]
```diff
@@ -405,10 +428,11 @@ DApp.prototype.schema = {
 
 /**
  * Deletes null or undefined dapp from transaction and validate dapp schema.
- * @implements {library.schema.validate}
- * @param {transaction} transaction
- * @return {transaction}
+ *
+ * @param {transaction} transaction - Description of the param
  * @throws {string} Failed to validate dapp schema.
```
[the remark this answers is not in the document]
Comment (Tschakki) [reply logic/dapp.js]: Should we create a new issue for this?

File: logic/multisignature.js (hunk 16 above)
[the remark this answers is not in the document]
Comment (Tschakki) [reply logic/multisignature.js]: seems like it still needs to be done.

File: logic/multisignature.js (hunk 13 above)
[the remark this answers is not in the document]
Comment (Tschakki) [reply logic/multisignature.js]: Sure, but in the cases where we provide this additional information, it doesn't really hurt, or not?

In my opinion, we should not use `{?number}`, because this is kind of the default, but if we want to specify, that a value cannot be null, `{!number}` can be a good way to be more precise

File: logic/peer.js (hunk 17 above)
[the remark this answers is not in the document]
Comment (Tschakki) [reply logic/peer.js]: As mentioned above, I think optional types should be default, so we don't need the brackets here.

File: logic/transaction_pool.js (hunk 8 above)
Comment (Tschakki) [reply logic/transaction_pool.js]: As discussed, we will open a new issue for standardizing typedefs in all files

File: logic/peer.js (hunk 17 above)
[the remark this answers is not in the document]
Comment (Tschakki) [reply logic/peer.js]: As discussed, we will do this in a seperate PR
